nncf/experimental/common/tensor_statistics/collectors.py

```diff
@@ -103,10 +110,11 @@ def __eq__(self, __o: object) -> bool:
             isinstance(__o, self.__class__)
             and self._reduction_axes == __o._reduction_axes
             and self._inplace == __o.inplace
+            and self._keep_empty_stats == __o._keep_empty_stats
         )
 
     def __hash__(self) -> int:
-        return hash((self.__class__.__name__, self.inplace, self._reduction_axes))
+        return hash((self.__class__.__name__, self.inplace, self._reduction_axes, self._keep_empty_stats))
```

Comment:
As hash is always overriding please update hashes for each reducer

Reply:
Outdated.